src/client.rs

```diff
@@ -81,12 +82,27 @@ impl ClientFlow {
         ));
 
         Self {
+            options,
             handle_generator: HANDLE_GENERATOR_GENERATOR.generate(),
             send_command_state,
             receive_state,
         }
     }
 
+    /// Changes the actual receive state codec with the
+    /// [`ResponseCodec`]'s one.
+    ///
+    /// The main use case of this function is straight after receiving
+    /// greeting, in order to replace the actual greeting codec by the
+    /// default response codec.
```

Comment:
You don't have to do this. `ClientFlow` already does this by default. See https://github.com/duesee/imap-flow/blob/e24687913ba0761235e6f1e863b17c732ff8b807/src/client.rs#L153-L157

Reply:
Oh, great! I missed it, thank you.